src/support.js

```diff
@@ -246,14 +246,40 @@ Module['registerFunctions'] = registerFunctions;
 #endif // RELOCATABLE
 #endif // EMULATED_FUNCTION_POINTERS
 
+#if WASM_BACKEND
+#if RESERVED_FUNCTION_POINTERS
+var jsCallStartIndex = {{{ JSCALL_START_INDEX }}};
+var jsCallSigOrder = {{{ JSON.stringify(JSCALL_SIG_ORDER) }}};
+var jsCallNumSigs = Object.keys(jsCallSigOrder).length;
+var functionPointers = new Array(jsCallNumSigs * {{{ RESERVED_FUNCTION_POINTERS }}});
+#else
+var jsCallStartIndex = 1;
+var functionPointers = new Array({{{ RESERVED_FUNCTION_POINTERS }}});
+#endif
+#else
+var jsCallStartIndex = 1;
 var functionPointers = new Array({{{ RESERVED_FUNCTION_POINTERS }}});
+#endif
 
+#if WASM_BACKEND
+function addFunction(func, sig) {
```

Comment:
Could maybe just always have the signature be `(func, sig)`, with a comment explaining that `sig` is only used `#if WASM_BACKEND`. JS functions don't mind being called with a mismatched number of arguments.

Reply:
Good point. Done.